### PR TITLE
Record view timing in instrumentation

### PR DESCRIPTION
**Pull request by asgrim**
https://trello.com/c/M1lit7Sf/8-add-laravel-view-instrumentation

### CODE REVIEW

File: src/Events/ScoutViewEngineDecorator.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * Get the evaluated contents of the view.
+     *
+     * @param string $path
+     * @param array $data
+     * @return string
+     */
+    public function get($path, array $data = [])
+    {
+        $this->agent->startSpan('viewEngine');
```
Comment (asgrim) [inline src/Events/ScoutViewEngineDecorator.php]: Happy to take guidance on what this span should be named, according to other Scout conventions...

File: src/Events/ScoutViewEngineDecorator.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        $return = $this->realEngine->get($path, $data);
+
+        $this->agent->stopSpan();
```
Comment (asgrim) [inline src/Events/ScoutViewEngineDecorator.php]: Also would you prefer to use `->instrument(..., function () ....)` for this?

File: src/Facades/ScoutApm.php [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+declare(strict_types=1);
 
 namespace Scoutapm\Laravel\Facades;
 
+use Closure;
 use Illuminate\Support\Facades\Facade;
+use Scoutapm\Events\Span\Span;
 
+/**
+ * @method static void connect()
```
Comment (asgrim) [inline src/Facades/ScoutApm.php]: I slipped these in to help consumers with static analysis when using Facades. Without these, most IDEs will complain that methods that don't exist are being called due to the hidden magic being used to make Facades work.

File: src/Events/ScoutViewEngineDecorator.php (hunk 1 above)
Comment (cschneid) [reply src/Events/ScoutViewEngineDecorator.php]: `View/file path snippet` ideally. So something like `View/users/index`

File: src/Events/ScoutViewEngineDecorator.php (hunk 2 above)
Comment (cschneid) [reply src/Events/ScoutViewEngineDecorator.php]: Doesn't really matter. I would however wrap this in a try/catch/finally block to be sure it closes even if the realEngine raises an exception.

File: src/Facades/ScoutApm.php (hunk 3 above)
Comment (cschneid) [reply src/Facades/ScoutApm.php]: 👍

**Comment by cschneid** [comment]
Awesome, I had assumed that wrapping the View engines with our own was what was going to need to happen, but didn't figure out how to make it happen. Glad it was easy enough.

**Pull request merged**